scico/numpy/_blockarray.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020-2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SPORCO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+r"""Block array class."""
+
+import inspect
+from functools import wraps
+from typing import Callable
+
+import jax
+import jax.numpy as jnp
+
+from jaxlib.xla_extension import DeviceArray
+
+from ._wrapped_function_lists import binary_ops, unary_ops
+
+
+class BlockArray(list):
+    """Block array class.
+
+    See :ref:`detailed BlockArray documentation <blockarray_class>`.
```

Comment:
@Michael-T-McCann: This is a bit ugly. Perhaps worth adding a brief summary here, in addition to the cross-reference to the full docs?

Reply:
Added.